### DISCUSSION TITLE
Incorrect DID returned?

### DISCUSSION THREAD

**Discussion by ghost** [bluesky-social/atproto#2909] [Bluesky Client App]
Hey everyone,

maybe one of you can assist here - I try to change my handle to my custom domain but get the error that an "Incorrect DID returned". I have never used the specific domain before and despite putting the current _atproto-handle in my DNS' providers settings (was CloudFlare before, then I switched DNS to my webspace provider - no change) I cannot get the handle to work.

<img width="549" alt="image" src="https://github.com/user-attachments/assets/9fa03980-a09d-484d-a95f-9112401fe8ef">

Is there anything cached from within BlueSky that I cannot force to refresh? Any other domain works without problems (or would work, as the system tells me that it is unable to resolve the handle. The problem occurs with my main domain only and there is no _atproto-entry in the DNS-settings - I tried this for ten days now and after migrating away from Cloudflare, this can't be a caching issue - can it?

Any help is appreciated here - thanks in advance!

**Comment by aidenfoxivey** [comment]
Got the same issue here with my PDS. Weird bug I think. I've tried refreshing my server's DNS cache itself, but it doesn't seem to make a difference.

`dig -t TXT +short _atproto.aidenfoxivey.com` returns `did=did:plc:uawamy37zg2n7ffoftnew5ee`

and 


`curl -s https://aidenfoxivey.com/.well-known/atproto-did` returns `did:plc:uawamy37zg2n7ffoftnew5ee`

**Comment by aidenfoxivey** [comment]
Hmm, okay, so I had success using the `goat` https://github.com/bluesky-social/indigo/tree/main/cmd/goat tool.

I did this:

<Install the goat tool.>

`goat account login -u did:plc:uawamy37zg2n7ffoftnew5ee -p <put your app password here>`

`goat account update-handle aidenfoxivey.com`

And it seemingly worked? When I go to my PDS now it now shows

```
root@ubuntu-4gb-fsn1-1:~# sudo pdsadmin account list
Handle            Email                   DID
aidenfoxivey.com  aiden@aidenfoxivey.com  did:plc:uawamy37zg2n7ffoftnew5ee
```

**Reply by ghost** [reply]
@aidenfoxivey You made my day - following your steps, I was able to update my handle for my domain on bsky.app. Can't believed it worked that flawlessly and with your support, my problem got just solved (while Bluesky's support still hasn't responded for triaging the problem further). THANKS!!! 🙏

**Reply by dsmithcloud** [reply]
I've tried to install goat but get this error:

build github.com/bluesky-social/indigo/cmd/goat: cannot load crypto/ecdh: malformed module path "crypto/ecdh": missing dot in first path element

**Reply by PovilasID** [reply]
> I've tried to install goat but get this error:
> 
> build github.com/bluesky-social/indigo/cmd/goat: cannot load crypto/ecdh: malformed module path "crypto/ecdh": missing dot in first path element
They are targeting fairly new go version, so for me easiest version is to run a temp docker go docker container:
```
docker run --rm -it golang:1.23
``` 
And then login and change inside it.

**Reply by fingerguns83** [reply]
Thank you for this! I've been fighting this for like a week.

**Reply by ghost** [reply]
YES! It works! You're a lifesaver, bro!

**Reply by aidenfoxivey** [reply]
Honestly I think it's just something related to the timeout of the app's cache for DNS queries being really long. I submitted a PR but I don't recall it getting traction. I don't blame them for being busy though lol.
